File: exploreArrays/C++/Ejercicio5.cpp
#include<vector>
#include<iostream>
#include<algorithm>
using namespace std;

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
      for(int i = m; i < nums1.size(); i++){
          nums1[i] = nums2[i-m];
      }
      sort(nums1.begin(), nums1.end());

    }
};
int main(){
    vector<int> vec1 = {1,2,3,0,0,0};
    vector<int> vec2 = {3,2,1};
    int m = 3;
    int n = 3;
    Solution sol;
    sol.merge(vec1, m, vec2, n);
    for(int i = 0; i < vec1.size(); i++){
        cout << vec1[i] << ","; 
    }


}

File: exploreHashmaps/Python/Ejercicio5.py
from typing import List

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        seen1 = set()
        seen2 = set()
        for element in nums1:
            seen1.add(element)
        for element in nums2:
            seen2.add(element)
        return list(seen1.intersection(seen2))
# Can be improved using just one set, then iterate the second array, 
# and if the element is in the first set, append to res, and delete from first set
sol = Solution()
print(sol.intersection([1,2,2,1], [2,2]))
print(sol.intersection([4,9,5], [9,4,9,8,4]))